Server es6 conversion

JadziaHax opened the pull request.

Commit:
Covert all server files to ES6

All files are now using ES6 (particularly module syntax). Transpiling of
the code before execution with node (or nodemon) via babel is required
now. Though, right now the transpilied server code is throwing an error.

Commit:
Add missing files from previous commit; Restore nominal functionality.

Restores full functionality of back-end after ES6 conversion including
user registration, user login and authorization check.

JadziaHax committed:
Merge branch 'master' into server-ES6-conversion

JadziaHax committed:
Restore concurrently to the npm start script

nguyenaiden commented on server/config/auth/passportConfig.js:
```diff
@@ -0,0 +1,29 @@
+jwtOptions.secretOrKey = 'JjS6E32hy1yBbhFN7uYI8t8IqpPy7xgw';
+
+const strategy = new JwtStrategy(jwtOptions, (jwtPayload, next) => {
+  console.log('payload received', jwtPayload);
```
Wild console.logs!

JadziaHax replied:
I know I told you this but for documentation's benefit I'm keeping console logs generally just because its useful for development, but we should remove them where it makes sense before finalizing our code.

nguyenaiden replied:
👍

nguyenaiden commented on server/config/auth/passportConfig.js:
```diff
@@ -0,0 +1,29 @@
+
+const jwtOptions = {};
+jwtOptions.jwtFromRequest = ExtractJwt.fromAuthHeader();
+jwtOptions.secretOrKey = 'JjS6E32hy1yBbhFN7uYI8t8IqpPy7xgw';
```
Put into bash profile

nguyenaiden reviewed (dismissed):
Take and make changes as you see fit.
👍

Commit:
Remove token secret and replace with environment variable.

Was publishing secret to public repository. Now the code uses dotenv npm
to process the key from a .env file that lives in the project's root
directory (this file is in gitignore, so make your own) that contains a
line that says:
secretOrKey=<INSERT SECRET HERE>
You have to create this file and assign your own secret where the
angle-bracketed phrase is.

nguyenaiden reviewed (approved):
👍
